src/paths_gen/PathFinder.java

```diff
@@ -121,4 +121,75 @@ private boolean isSamePair(Path tic, Path tac) {
     private String pairDescriptor(Vertex src, Vertex dst) {
         return src.name() + ":" + dst.name();
     }
+
+    public List<Path> allPathsComputationByVertex(Vertex src, Vertex dst) {
+        ArrayList<Path> result = new ArrayList<>();
+        Set<String> alreadyFoundPairs = new HashSet<>();
+        // N = 1 hop
+        ArrayList<Path> oneHopPaths = computeOneHopPaths(src);
+        result.addAll(oneHopPaths);
+        ArrayList<Path> previouslyFoundPaths = oneHopPaths;
+        // N > 1 hop
+        int nPairs = graph.getVertices().size() * (graph.getVertices().size() - 1);
+        while (!previouslyFoundPaths.isEmpty()) {
+            ArrayList<Path> valid = advanceOneHopNoMinimal(previouslyFoundPaths, alreadyFoundPairs);
+            for(Path path : valid){
+                if(path.dst().equals(dst)){
+                    result.add(path);
+                }
+            }
+            previouslyFoundPaths = valid;
+            previouslyFoundPaths.removeAll(result);
+        }
+        Collections.sort(result);
+        return result;
+    }
+
+    private ArrayList<Path> computeOneHopPaths(Vertex src) {
+        ArrayList<Path> result = new ArrayList<>();
+        for (Edge e : graph.adjunctsOf(src)) {
+            Vertex dst = e.destination();
+            if (restrictions.turnIsForbidden(src, 'I', graph.adjunct(src, dst).color())) {
+                continue;
+            }
+            Path p = new Path();
+            p.add(src);
+            p.add(dst);
+            result.add(p);
+        }
+        return result;
+    }
+
+    private ArrayList<Path> advanceOneHopNoMinimal(ArrayList<Path> previouslyFoundPaths, Set<String> alreadyFoundPairs) {
+        ArrayList<Path> result = new ArrayList<>();
+        for (Path p : previouslyFoundPaths) {
+            result.addAll(advanceOneHopNoMinimal(p, alreadyFoundPairs));
+        }
+        for (Path p : result) {
+            alreadyFoundPairs.add(pairDescriptor(p.src(), p.dst()));
+        }
+        return result;
+    }
+
+    private ArrayList<Path> advanceOneHopNoMinimal(Path p, Set<String> alreadyFoundPairs) {
```

Comment:
Rename it to something as 'oneHopMorePathsFrom' and return a 'List' instead of 'ArrayList'.